resource-sequence.puml

```diff
@@ -70,53 +72,69 @@ loop For each resource subscriber
+op -> OIM: Op Request
+OIM -> DSS: PUT /dss/v1/operational_intent_references/{entityId}  (Proposed state)
+note over DSS
+Does not check any OVNs here.
+All operation intents accepted.
+end note
+OIM <-- DSS: subscribers to notify
+loop For all subscribers
+    OIM -> DCB: POST /dcb/v1/operational_intents (includes OVN)
+    DCB -> DCB: Adjust demand/capacity values
+    OIM -> DCB2:  POST /dcb/v1/operational_intents (includes OVN)
+    DCB2 -> DCB2: Adjust demand/capacity values
+end
 
+== A new subscriber joins == 
 DCB2 -> DSS: PUT /DSS/v1/subscriptions/resource-subscription/
 DCB2 <-- DSS: resource references
-
 loop For each resource reference received
     note over DCB2
     Get all resource details.
     end note
     DCB2 -> RIS: GET /v1/resource-details/{resource-id} (includes OVN)
     DCB2 <-- RIS: resource details (everything)
 end
-
-
-== OIM2 submits an operation to the DSS == 
-OIM2 -> DSS: PUT /dss/v1/operational_intent_references/{entityId}  (Proposed state)
-note over DSS
-Does not check any OVNs here.
-All operation intents accepted.
-end note
-OIM2 <-- DSS: subscribers to notify
-loop For all subscribers
-    ' OIM2 -> RIS: /psu/v1/operational_intents (includes OVN)
-    OIM2 -> DCB: /psu/v1/operational_intents (includes OVN)
-    DCB -> DCB: Adjust demand/capacity values
-    OIM2 -> DCB2:  /psu/v1/operational_intents (includes OVN)
-    DCB2 -> DCB2: Adjust demand/capacity values
+DCB2 -> DSS: PUT /DSS/v1/subscriptions/operation-subscription/
+DCB2 <-- DSS: operation references
+loop For each operation reference received
+    note over DCB2
+    Get all resource details.
+    end note
+    DCB2 -> OIM: GET /v1/operation_details/{operation-id} (includes OVN)
+    DCB2 <-- OIM: operation details (everything)
 end
 
-== OIM wants to transition operation to Accepted state (OVNs required) ==
-' OIM -> DCB: Get available resources
-' OIM <-- DCB: Return all resources, OVNs, avaialble time ranges
-
 
-OIM -> DCB: Submit proposed intent
+== OIM does DCB check and transitions operation to Accepted state (OVNs required) ==
+OIM -> DCB: POST /dcb/v1/imbalance/intent
 OIM <-- DCB: Yes/no, OVNs for all resources, available time ranges
 
-OIM -> DSS: PUT /dss/v1/operation_intent_references/{entityId}/{ovn}\n(state = Accepted)
-DSS -> DSS: Confirms OVNs are correct
-OIM <-- DSS: operation intent reference with subscribers
-loop For all subscribers
-    OIM -> DCB2: PUT /psu/v1/operational_intents
-    OIM -> DCB: PUT /psu/v1/operational_intents
+alt If imbalance detected
+    note over OIM: Do not update the operation
+else If no imbalance detected
+    OIM -> DSS: PUT /dss/v1/operation_intent_references/{entityId}/{ovn}\n(state = Accepted)
+    DSS -> DSS: Confirms OVNs are correct
+    OIM <-- DSS: operation intent reference with subscribers
+    loop For all subscribers
+        OIM -> DCB2: POST /dcb/v1/operational_intents
+        OIM -> DCB: POST /dcb/v1/operational_intents
+    end
 end
 
+==Telemetry Sharing==
```

Comment:
I don't understand what exactly we're trying to do in this section.  

I'm assuming CM is the conformance monitoring for OIM and CM2 is for a different OIM2 (that probably uses DCB2) that we haven't fully shown.  

In line 129, are we planning to define a OIM -> CM specific message that is different than the operational_intents to provide details about the flight criteria to monitor conformance against?  If this is the same as the PSU->PSU operational_intents message, then why not just have CM subscribe to DSS for operations? If it is different content, we might want to call it something different here. 

I think we're trying to show both a CM receiving telemetry from an aircraft and doing conformance monitoring and also trying to show how two CMs would get telemetry from each other.  If that is the case, we should split this into a couple sections to make it more clear what we're trying to show.

Reply:
All good points. My only reason for the OIM->CM operation intent was that if they are separate services, the CM is not going to know about the operation intent. But you are correct that it could subscribe like everyone else. This may need some discussion with the group though to confirm. I'll create a separate section for conformance monitoring. There's more content needed there too I think.